### PR TITLE
Fix grid ids in migrated JS

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.5.x
| Description?  | Fix Grid ids in js. It's related to #10569 
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Grid ("Show SQL query", "Refresh list" & etc.) and Bulk ("Delete selected" & etc.) actions of migrated list should work as they did before. 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10656)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

File: admin-dev/themes/new-theme/public/catalog_product.bundle.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
...
....prop("checked",!0)})}},{key:"enableSpecificPriceFieldIfEligible",value:function(e){var i=this.getPrefixSelector(e);o(i+"sp_price").prop("disabled",o(i+"leave_bprice").is(":checked")).val("")}},{key:"openEditPriceModalAndLoadForm",value:function(e){var i=this,t=o("#js-specific-price-list").data("actionEdit").replace(/form\/\d+/,"form/"+e);o("#edit-specific-price-modal").modal("show"),this.editModalIsOpen=!0,o.ajax({type:"GET",url:t}).done(function(t){i.insertEditSpecificPriceFormIntoModal(t),o("#edit-specific-price-modal-form").data("specificPriceId",e),i.configureEditPriceFormInsideModalBehavior()}).fail(function(e){showErrorMessage(e.responseJSON)})}},{key:"closeEditPriceModalAndRemoveForm",value:function(){o("#edit-specific-price-modal").modal("hide"),this.editModalIsOpen=!1,o("#edit-specific-price-modal-form").empty()}},{key:"insertEditSpecificPriceFormIntoModal",value:function(e){var i=o("#edit-specific-price-modal-form");i.empty(),i.append(e)}},{key:"getProductId",value:function(){return o("#form_id_product").val()}},{key:"getPrefixSelector",value:function(e){return 1==e?"#"+this.prefixCreateForm:"#"+this.prefixEditForm}}]),e}();i.a=a},409:function(e,i,t){e.exports=t(197)}});
```
Comment (LouiseBonnard) [inline admin-dev/themes/new-theme/public/catalog_product.bundle.js]: It seems that there are a few wordings here... should I check them and indicate domains?

Comment (sarjon) [reply admin-dev/themes/new-theme/public/catalog_product.bundle.js]: nope, there are no wording in this PR.

Comment (LouiseBonnard) [reply admin-dev/themes/new-theme/public/catalog_product.bundle.js]: Nice, thanks for the answer!

**Comment by sarjon** [comment]
@PierreRambaud any ideas whats wrong with unit tests? :/

**Comment by PierreRambaud** [comment]
@sarjon Done mate ;)

**Comment by marionf** [comment]
Hello @sarjon 

I have an exception when I try to delete in bulk action on the Traffic & SEO page

capture d ecran_318

File: src/Core/Grid/Presenter/GridPresenter.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function present(GridInterface $grid)
         $searchCriteria = $grid->getSearchCriteria();
         $data = $grid->getData();
         $presentedGrid = [
-            'id' => $definition->getId(),
+            'id' => strtolower($definition->getId()),
```
Comment (sarjon) [inline src/Core/Grid/Presenter/GridPresenter.php:58]: i dont like this workaround, but its related to this issue https://github.com/PrestaShop/PrestaShop/issues/10412 :(

**Comment by matks** [comment]
I agree with this change but this is a major technical setting so we need eternoendless to agree with it.
 
@eternoendless can you review https://github.com/PrestaShop/PrestaShop/issues/10412 and sarjon latest commit to fix it ?

**Comment by marionf** [comment]
It doesn't fix the issues, maybe an assets problem ?

**Comment by sarjon** [comment]
> maybe an assets problem

yes, im waiting for feedback on this solution before rebasing & building assets. :/

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@sarjon Solution approved by eternoendless . We're good here ! Please rebase & rebuild assets

**Comment by sarjon** [comment]
rebase done.

i noticed that we need to change more things in order to fix everything. :/ what if we keep grid ids as they where before and only update how hooks are executed? For example instead of this:

```php
$this->hookDispatcher->dispatchWithParameters('action' . $definition->getId() . 'GridDefinitionModifier', [
    'definition' => $definition,
]);
```
we keep same ids and do this:

```php
$this->hookDispatcher->dispatchWithParameters('action' . SomeHelper::toCamelCase($definition->getId()) . 'GridDefinitionModifier', [
    'definition' => $definition,
]);
```

so key `email_logs` would become `EmailLogs`?

@matks wdyt?

**Review by matks — APPROVED** [review]

**Comment by eternoendless** [comment]
Broken tests unrelated to PR

**Comment by eternoendless** [comment]
Thank you @sarjon

**Pull request merged**